From: Chengguang Xu <cgxu519@gmx.com>
Date: Fri, 9 Mar 2018 15:12:40 +0800
Subject: ceph: filter out used flags when printing unused open flags
Git-commit: 51b10f3fe446f536b6edf90ce6941882033dd93b
Patch-mainline: v4.17-rc1
References: bsc#1089115

Filter out used access mode flags when printing unused open flags.

Signed-off-by: Chengguang Xu <cgxu519@gmx.com>
Reviewed-by: "Yan, Zheng" <zyan@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/file.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -29,6 +29,8 @@ static __le32 ceph_flags_sys2wire(u32 fl
 		break;
 	}
 
+	flags &= ~O_ACCMODE;
+
 #define ceph_sys2wire(a) if (flags & a) { wire_flags |= CEPH_##a; flags &= ~a; }
 
 	ceph_sys2wire(O_CREAT);
